base/generator.jl

```diff
@@ -20,6 +20,12 @@ function next(g::Generator, s)
     g.f(v), s2
 end
 
+"""
+    generate(f, c...) -> iterator
+
+Apply `f` to each element of `c`.
+For multiple collection arguments, apply f elementwise.
+"""
 generate(f, c...) = Generator(f, c...)
```

Comment:
Why wouldn't you want to allow `generate` to return any one of `FallbackGenerator`, `GeneratorSpecialisedForVectors`, `GeneratorOptimisedForInputOfKnownLength`, `GeneratorOptimisedForSingleInputCollection`, etc, etc. ?

```julia
julia> typeof(zip([1]))
Base.Zip1{Array{Int64,1}}

julia> typeof(zip([1],[2]))
Base.Zip2{Array{Int64,1},Array{Int64,1}}

julia> typeof(zip([1],[2],[3]))
Zip{Array{Int64,1},Base.Zip2{Array{Int64,1},Array{Int64,1}}}
```